### DISCUSSION TITLE
Use a Shelly as a security system in HomeKit

### DISCUSSION THREAD

**Discussion by WaldperlachFabi** [mongoose-os-apps/shelly-homekit#1504] [Ideas]
I would like to use a Shelly plug s as a HomeKit security system. So is it possible to add the security system function for my Shelly with a Softwareupdate? I’d like to do something with something like [this](https://www.amazon.de/gp/product/B013S71T4C/ref=as_li_qf_asin_il_tl?ie=UTF8&tag=iphone019-21&creative=6742&linkCode=as2&creativeASIN=B013S71T4C&linkId=3ce028c95b2a293eb8527a886e860019)

**Comment by timoschilling** [comment]
What does you concrete want to control with the Shelly?

**Comment by WaldperlachFabi** [comment]
> What does you concrete want to control with the Shelly?

I want to control an alarm bell and have an icon like this  in Home. I would only another icon. And a „Switch“ in HomeKit. When I turn the switch on the alarm bell should go on or if I put the status on „off“ there should be a sound and if I put it on „away“ there should be a sound. ☻ 
![B5A8BF81-C1B0-4F0B-AA10-CE1B351352DE](https://user-images.githubusercontent.com/80318133/110782365-56891980-8267-11eb-9e11-676b62301022.jpeg)

**Comment by WaldperlachFabi** [comment]
![C3897043-4763-4FED-BB89-3B1E58BAA39C](https://user-images.githubusercontent.com/80318133/110783696-10cd5080-8269-11eb-8777-1efbe8cae356.jpeg)

**Comment by WaldperlachFabi** [comment]
<img width="1112" alt="IMG_2109" src="https://user-images.githubusercontent.com/80318133/110925585-adecbf80-8323-11eb-87dc-036ef7dea27d.png">

**Comment by AG-dingoant** [comment]
Hi all,
I am interested as well to have the security option in the shelly. I have a bentel alarm system and I can get signals from it. I will connect the signal to the switch of a shelly module.
It will allow me to know if the system is activated or not. If it is activated, than I would like to have all the lights switched off. 
If the alarm rings, than all the lights have to be switched on.
I can set the shelly as an switch but in homekit it will not be displayed as an alarm system.

**Comment by Chpedro93** [comment]
Good morning,
Any news regarding this? It will be implemented in the future updates? 
Thanks.

**Comment by 5xPa** [comment]
Thanks for this 
I am looking for the security system is “armed” meaning the system is monitoring the house. Which as described above would then switch all the lights and heating off. I have a uni set as a contact to accomplish this. When I return home and disarm the system, heating on.

Another uni is used when the system is triggered and bell ringing sends a push notification, this is also set as a contact.

Has the updates been added to a beta I would be delighted to test the update. 
I have installed the beta 2.11.0-beta3 for 35 days and expierenced  no problems.
Thank you for the excellent upgrades

**Comment by iMakit** [comment]
Hi all, I'm trying to use Shelly 1 connected to my "old" alarm hub to set alarm ON and OFF, at the moment I set in the hub one key to be used as dry contact to arm and disarm, I'm setting hap service type as switch and toggle, any suggestion to be used as alarm type?
something like abode system also if at the moment I can use only one
<img width="1021" alt="Schermata 2022-06-26 alle 18 51 39" src="https://user-images.githubusercontent.com/108232887/175825614-5b94c5e5-2213-49f5-9880-79b0e8be3113.png">

I just saw the new firmware is compatible with Shelly UNI, tomorrow I'll get one for testing if it can manage 2 states as On and away mode, any suggestion on how it may be configured? 

And thanks to all of you for amazing job and upgrades

**Comment by iMakit** [comment]
Hi tried with Shelly UNI but it's like to have 2 separated switch, is it possible to have something like the alarm to cutaway and nigh mode for example? Something like Abode system above 
I saw the trigger documentation [here](https://developer.apple.com/documentation/homekit/hmcharacteristictypesecuritysystemalarmtype) but not sure how it can be used to arm security alarm. Did someone tried to connect alarm hub that may help? Thanks :)

**Comment by iMakit** [comment]
I made a small improvement to the alarm system, I configured a shelly UNI as 2 switches and momentary impulse and one entry of the central station as key point, that means that with a shelly one impulse set the alarm ON and one again will set it OFF.
I also added an automation in HomeKit put the alarm ON when the last person goes away and set OFF when the first person come back. Would be great if it can be managed in security instead of switch with possibility to set TOTAl PARTIAL and OFF alarm.
